cmd/osbuild-composer/main.go

```diff
@@ -5,8 +5,8 @@ import (
 	"log"
 	"os"
 
+	"github.com/osbuild/osbuild-composer/internal/dnfadapter"
```

Comment:
The name `rpmmd` comes from [RPM Metadata](https://en.opensuse.org/openSUSE:Standards_Rpm_Metadata), the spec which defines how repository metadata is stored. The idea was that the package to provide an API to download and query such metadata, but it grew into a bit of a mess indeed.

I agree that `rpmmd` is clunky to say and write, but it describes *what* the package does. `dnfadapter` describes *how* it's being done. I think the former is preferable, because it allows switching the *how*.

Reply:
I chose the approach of mocking `dnf-json` because it also tests the marshal/unmarshal bit, which seemed a bit fragile to me - passing data from one language to another can be always tricky. Also with mocks in json format we have quite nice documentation of possible outputs of `dnf-json`. About the added complexity - I find both solutions very similar in terms of complexity.

About the `rpmmd` - I know what you mean with "*how* it's being done". Although, instead of `DNFAdapter` calling the type `RPMMD` (all caps, I guess?) sounds weird to me. Calling method `Depsolve` on object of name "metadata" is weird, metadata sounds like static blob, not something which can do depsolving.

DNF is a package manager. What about calling the struct PackageManager? It explains "what it does" and sounds like something able to depsolve and dump packages.

Reply:
The package name can still be rpmmd, I haven't been able to come up with something better.

Reply:
Let's leave the name rpmmd for now, and we can discuss renaming separately.

Reply:
It is true that you'll get more testing by mocking the backend rather than the golang package. However, I think we should avoid this sort of accidental/secondary testing, it leads to unclear tests in the long run. I'd agree with Lars that mocking the golang package is the way to go, and let these tests only be about testing the weldr API.

This is, by the way, the same reason we try to only have one API test at a time, rather than combining several. It means the code is more verbose, but I think it makes it clearer what exactly it is we are testing and make it easier to know what we have and haven't covered and how to easily add to the tests in the future.